packages/dom-adapters/src/FormattingAdapter/index.ts

```diff
@@ -171,9 +173,72 @@ export class FormattingAdapter {
         return;
       }
 
-      const inlineElement = tool.createWrapper(event.detail.data.data);
+      const affectedFragments = this.#model.getFragments(blockIndex, dataKey, ...textRange);
 
-      surround(inlineElement, input, textRange);
+      let lowerBoundary = textRange[0];
+      let upperBoundary = textRange[1];
+
+      for (const fragment of affectedFragments) {
+        lowerBoundary = Math.min(lowerBoundary, fragment.range[0] ?? 0);
```

Comment:
Model returns fragments in order, so the left boundary will always be the first fragment